### ISSUE TITLE
changes to locations do not always trigger reindex and reprPoint recalculation (3 points)

### ISSUE THREAD

**Issue by paregorios** [isawnyu/pleiades-gazetteer#438]
Logged-in reviewers and other users don't always see the spatial results of changes they make to places in a timely manner. Also, retraction and publication of locations don't seem to always trigger a spatial re-index of the containing place, so that users are not getting the latest reprPoint values and the place map doesn't reflect the current state and position(s) of the locations.

The rules should be:

1. Only published locations should be used for reprPoint calculations.
2. Recalculate reprPoint AND spatially re-index the place whenever any of the following happens to a contained location:
   a. published
   b. retracted
3. Whenever reprPoint is recalculated or the place is reindexed, purge from cache all serializations associated with this place (especially the GEOJSON that place_map.js uses internally).

It would also be helpful if, for authenticated users, we do whatever we can to keep their browser from hanging on to an expired value after they make edits like the above.

**Comment by cguardia** [comment]
@paregorios is there an example that you could point me to?

**Comment by paregorios** [comment]
@cguardia I have cleaned up (reindexed) all the cases I previously came across. I have tried this morning to reproduce the error, but I have been unsuccessful. Let's defer further work on this ticket and move on to other things for now.

**Comment by paregorios** [comment]
I think I have been able to create a case of this on staging: https://pleiades.jazkarta.com/places/520977

Here's what I did:

1. Changed the connection types on the two inbound connections from "connection" to "communicates with".
2. Retracted the previously published location.

Problems:

1. I see no change in the reprPoint calculations although examination of the code leads me to believe that it should have been recalculated on the basis of the two connections (so I'm suspicious that no sort of re-index/recalc occurred).
2. The map still shows (even in a new anonymous browser) both the orange reprPoint icon (expected) and the blue location icon (erroneous).

Putting this back in the project if we have time to address.
